hledger-lib/Hledger/Read/Common.hs

```diff
@@ -1011,17 +1001,15 @@ data AmbiguousNumber = AmbiguousNumber DigitGrp Char DigitGrp
 -- | Description of a single digit group in a number literal.
 -- "Thousands" is one well known digit grouping, but there are others.
 data DigitGrp = DigitGrp {
-  digitGroupLength :: !Int,    -- ^ The number of digits in this group.
-  digitGroupNumber :: !Integer -- ^ The natural number formed by this group's digits.
+  digitGroupLength :: !Word,    -- ^ The number of digits in this group.
```

Comment:
why the change to Word ? to allow groups of more than 255 digits in the integral part ?

Reply:
It was just to get around the need to do overflow checks in the Semigroup instance of `DigitGrp`.